docs/user/external-database.md

```diff
@@ -0,0 +1,256 @@
+3. **Run database migrations** to set up the schema
+4. **Set up permissions** for the application user (if admin credentials are provided)
+
+### User Management Options
+
+#### Option 1: Manual User Setup (Recommended for Production)
+
+- Create users manually using the SQL script in the "Database Setup" section
+- Only provide `password` and `migrationPassword` in your configuration
+- FlightCtl will use existing users and skip user creation
+
+#### Option 2: Automatic User Creation
+
+- Provide `masterPassword` (admin credentials) in your configuration
+- FlightCtl will automatically create missing users and set up permissions
+- Useful for development or if you want FlightCtl to manage database users
+
+The migration process can use up to three different users:
+
+- **Admin user** (`masterUser`): Creates other users and databases (optional)
+- **Migration user** (`migrationUser`): Runs schema migrations and DDL operations (required)
+- **Application user** (`user`): Used by FlightCtl services for data operations (required)
+
+## Troubleshooting
+
+### Connection Issues
+
+1. **Check network connectivity**:
+
+   ```bash
+   # From your Kubernetes cluster or container environment
+   telnet your-postgres-hostname.example.com 5432
+   ```
+
+2. **Verify PostgreSQL configuration**:
+   - Ensure `listen_addresses` includes your FlightCtl network
+   - Check `pg_hba.conf` for proper authentication rules
+   - Verify firewall rules allow connections
+
+3. **Check logs**:
+
+   ```bash
+   # Kubernetes
+   kubectl logs -f deployment/flightctl-api
+   kubectl logs -f job/flightctl-db-migration-<revision>
+   
+   # Podman
+   podman logs flightctl-api
+   ```
+
+### Permission Issues
+
+If you see permission errors, ensure:
+
+1. The migration user has CREATE privileges on the database and schema
+2. The application user has appropriate data access privileges
+3. The admin user can create other users
+
+### SSL/TLS Issues
+
+For SSL connections, you may need to:
+
+1. Add SSL configuration to the database connection string
+2. Mount SSL certificates in your containers
+3. Configure `sslmode`, `sslcert`, `sslkey`, and `sslrootcert` parameters
+
+## Security Considerations
+
+1. **Use strong passwords** for all database users
+2. **Enable SSL/TLS** for database connections in production
+3. **Restrict network access** to your database instance
+4. **Use secrets management** instead of plain-text passwords in configuration
+5. **Regular backup** your external database
+6. **Monitor database performance** and set up appropriate alerts
+
+## Backup and Recovery
+
+When using an external database, you are responsible for:
+
+1. **Database backups**: Set up regular automated backups
+2. **Point-in-time recovery**: Configure WAL archiving if needed
+3. **High availability**: Configure replication or clustering
+4. **Monitoring**: Set up database monitoring and alerting
+
+FlightCtl does not manage backups for external databases.
```

Comment:
This is true for the internal DB as well - remove

Reply:
removed